moveit_ros/perception/handeye_calibration_target/src/handeye_aruco_target.cpp

```diff
@@ -0,0 +1,258 @@
+    markers_x_ = markers_x;
+    markers_y_ = markers_y;
+    marker_size_ = marker_size;
+    separation_ = separation;
+    border_bits_ = border_bits;
+
+    const auto& it = dict_map_.find(dictionary_id);
+    if (it != dict_map_.end())
+      dict_ = it->second;
+    else
+    {
+      ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid dictionary name.");
+      return false;
+    }
+  }
+  else
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid target intrinsic params.");
+    return false;
+  }
+
+  return true;
+}
+
+bool HandEyeArucoTarget::setTargetDimension(const double& marker_size, const double& marker_seperation)
+{
+  if (marker_size != marker_size_real_ || marker_seperation != marker_seperation_real_)
+    ROS_DEBUG_STREAM_NAMED(LOGNAME, "Set target real dimensions: "
+                                        << "\n"
+                                        << "marker_size " << std::to_string(marker_size) << "\n"
+                                        << "marker_seperation " << std::to_string(marker_seperation) << "\n");
+
+  if (marker_size > 0 && marker_seperation > 0)
+  {
+    std::lock_guard<std::mutex> lck(aruco_mtx_);
+    marker_size_real_ = marker_size;
+    marker_seperation_real_ = marker_seperation;
+    return true;
+  }
+  else
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid target real dimensions.");
+    return false;
+  }
+}
+
+bool HandEyeArucoTarget::createTargetImage(cv::Mat& image)
+{
+  cv::Size image_size;
+  image_size.width = markers_x_ * (marker_size_ + separation_) - separation_ + 2 * separation_;
+  image_size.height = markers_y_ * (marker_size_ + separation_) - separation_ + 2 * separation_;
+
+  try
+  {
+    // Create target
+    cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(dict_);
+    cv::Ptr<cv::aruco::GridBoard> board =
+        cv::aruco::GridBoard::create(markers_x_, markers_y_, float(marker_size_), float(separation_), dictionary);
+
+    // Create target image
+    board->draw(image_size, image, separation_, border_bits_);
+  }
+  catch (cv::Exception& e)
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Aruco target image creation exception: " << e.what());
+    return false;
+  }
+
+  return true;
+}
+
+bool HandEyeArucoTarget::detectTargetPose(cv::Mat& image)
+{
+  std::lock_guard<std::mutex> lck(base_mtx_);
+  try
+  {
+    // Detect aruco board
+    aruco_mtx_.lock();
+    cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(dict_);
+    cv::Ptr<cv::aruco::GridBoard> board =
+        cv::aruco::GridBoard::create(markers_x_, markers_y_, marker_size_real_, marker_seperation_real_, dictionary);
+    aruco_mtx_.unlock();
+    cv::Ptr<cv::aruco::DetectorParameters> params_ptr(new cv::aruco::DetectorParameters());
+#if CV_MAJOR_VERSION == 3 && CV_MINOR_VERSION == 2
+    params_ptr->doCornerRefinement = true;
+#else
+    params_ptr->cornerRefinementMethod = cv::aruco::CORNER_REFINE_NONE;
+#endif
+
+    std::vector<int> ids;
```

Comment:
What ids?

Reply:
This is the ids of aruco markers. The name has been changed to more obvious.